vector/src/main/java/im/vector/app/features/onboarding/ftueauth/SplashCarouselStateFactory.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.content.Context
+import androidx.annotation.AttrRes
+import im.vector.app.R
+import im.vector.app.core.epoxy.charsequence.EpoxyCharSequence
+import im.vector.app.core.epoxy.charsequence.toEpoxyCharSequence
+import im.vector.app.core.resources.StringProvider
+import im.vector.app.features.themes.ThemeUtils
+import me.gujun.android.span.span
+import javax.inject.Inject
+
+class SplashCarouselStateFactory @Inject constructor(
+        private val context: Context,
+        private val stringProvider: StringProvider
+) {
+
+    fun create() = SplashCarouselState(listOf(
```

Comment:
the state creation has been extracted out to a dedicated factory as we need to apply some span logic to the terminating `.`